build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -473,7 +473,10 @@ message Action {
   // as long as the server will let it. The server SHOULD impose a timeout if
   // the client does not specify one, however, if the client does specify a
   // timeout that is longer than the server's maximum timeout, the server MUST
-  // reject the request.
+  // reject the request. The method of timekeeping used to apply the timeout
+  // MUST be consistent with what is used to compute
```

Comment:
MUST -> SHOULD? I'm not sure what material harm it would cause if a server is inconsistent here. E.g. if a server applied hidden padding to the timeout in some cases because it's expecting unknown delays (+100ms to all timeouts in case a live migration happens, say), but reports the simple delta to users, that seems fine enough to me?

Reply:
Buildbarn supports a feedback-driven analysis process to automatically determine the optimal worker size for an action:

https://www.youtube.com/watch?v=3eKVBwlAHsk

One of the things it wants to prevent is causing excessive stalls in case an action behaves exceptionally poorly on a worker that is too small. For example, an action that succeeds in 10 seconds on a 4-CPU system, but takes more than 5 minutes on a 1-CPU system. To solve that, it uses samples from previous executions to derive a safe execution timeout. For this to work correctly, the method of timekeeping must be consistent.

Right now this feature is integrated into the scheduler, so in a certain way an implementation detail of the system. That said, there is no reason why certain forms of analysis can't be done on the client side if you wanted to. That's why I picked the word MUST.